hana/monitoring.sls

```diff
@@ -0,0 +1,54 @@
+{%- from "hana/map.jinja" import hana with context -%}
+
+include:
+  - hana.install_pydbapi
+
+prometheus-hanadb_exporter:
+  pkg.installed:
+  - retry:
+      attempts: 3
+      interval: 15
+
+hanadb_exporter_logging_configuration:
+  file.managed:
+    - name: /etc/hanadb_exporter/logging_config.ini
+    - source: /usr/etc/hanadb_exporter/logging_config.ini
+    - require:
+      - prometheus-hanadb_exporter
+
+hanadb_exporter_metrics_configuration:
+  file.managed:
+    - name: /etc/hanadb_exporter/metrics.json
+    - source: /usr/etc/hanadb_exporter/metrics.json
+    - require:
+      - prometheus-hanadb_exporter
+
+{% for node in hana.nodes if node.host == grains['host'] %}
```

Comment:
I think we need to change the strategy here. We cannot just copy the configuration file in all the nodes and start the exporter, as some of them don't have the `exporter` data in the pillar file, and the template cannot be rendered. In fact, some of the entries cannot have the exporter running (the secondary machine for example).

This is what I would do:
- Create the config file only the nodes where the `exporter` entry exist
- Start the exporter only for the previous condition too
- We need to replicate this config file in any other node that shares the `sid` and `instance number`, but without starting the exporter. We might only want this if `ha_enabled` param is true.

Have in mind that when we export metrics from a HANA env with system replication, we only need metrics from the primary, and not the secondary. We need to play with this.

Reply:
The problem I have is that `hana.nodes` doesn't actually model the nodes, but it models HANA instances.
So in the example deployment, for instance, we have 3 elements in the pillar, but only 2 actual nodes.
If I try to replicate the primary config in all the nodes, I would get a third attempt at creating the same configuration in node 2, because of the third element in the array for the QA instance.
Should we rethink this completely or find a way around it?

Reply:
That's why I suggested to copy the configuration files only if the `sid` and `instance number` match. It's not possible to have 2 sap instances in the same node with that data, so we shouldn't never have the scenario you comment.

For example:
- I have 3 sap instances defined as in our example.
- I only define the exporter in the first primary node (and optionally in the QAS, this depends on the user desires)
- I don't need to define the exporter data in the secondary, as we cannot start the exporter there
- In this case only first node sid/instance number and secondary sid/instance number match, so we would create a copy of the exporter config file there. The state would be executed only twice, it's not possible to have duplicated things here